### PR TITLE
Authentication (kind of sort of)

### PR DESCRIPTION
**Pull request by avand**
Implementing a very basic authentication scheme:

* User sign in page, login/logout buttons, profile page link in header
* Two migrations: adding email and password to the users table
* Wrote some specs (see commit history)
* Consistent alert/notices in the layout

### CODE REVIEW

**Review by ryfauber — APPROVED** [review]
Looks good!

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class SessionsController < ApplicationController
+    # Sign in form
+    def new
+    end
+    
+    # Submitting the sign in form
+    # TODO case insensitive email look ups
```
Comment (avand) [inline app/controllers/sessions_controller.rb:7]: @ryfauber I know you're big on case insensitivity—I promise a pull request soon ;)

Comment (ryfauber) [reply app/controllers/sessions_controller.rb:7]: you better!

File: Gemfile [hunk 2]
```diff
@@ -26,7 +26,7 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 gem 'bootstrap', '~> 4.1.3'
 
 # Use ActiveModel has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
+gem 'bcrypt', '~> 3.1.7'
```
Comment (avand) [inline Gemfile]:
```suggestion
gem 'bcrypt', '~> 3.1.8'
```

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (mpatterson47) [reply app/controllers/sessions_controller.rb:7]: Insensitivity never makes sense.

**Comment by mpatterson47** [comment]
Awesome. I love authentication. :)

**Review by mpatterson47 — APPROVED** [review]

**Pull request merged**